lib/datasets.go

```diff
@@ -670,12 +670,16 @@ func (r *DatasetRequests) Validate(p *ValidateDatasetParams, errors *[]jsonschem
 		return r.cli.Call("DatasetRequests.Validate", p, errors)
 	}
 
-	if err = DefaultSelectedRef(r.repo, &p.Ref); err != nil {
+	if err = DefaultSelectedRef(r.repo.Repo, &p.Ref); err != nil {
 		return
 	}
 
-	if p.Ref.IsEmpty() && p.Data == nil {
-		return fmt.Errorf("either data or a dataset reference is required")
+	// TODO: restore
```

Comment:
Please expand on this comment. What is different about this commented out code from the code being added? Is it the ability to use a URL? If so, make the comment about that. Alternatively, just delete it instead of disabling it, and add a TODO like "allow validation from a URL"

Also, use tabs instead of spaces.